diff --git a/node_modules/@graphprotocol/graph-cli/src/command-helpers/compiler.js b/node_modules/@graphprotocol/graph-cli/src/command-helpers/compiler.js
index e696aa1..8a91c69 100644
--- a/node_modules/@graphprotocol/graph-cli/src/command-helpers/compiler.js
+++ b/node_modules/@graphprotocol/graph-cli/src/command-helpers/compiler.js
@@ -3,6 +3,11 @@ const ipfsHttpClient = require('ipfs-http-client')
 const toolbox = require('gluegun/toolbox')
 const Compiler = require('../compiler')
 
+const infuraSecret = process.env.INFURA_SECRET
+const headers = infuraSecret ? {
+  authorization: `Basic ${Buffer.from(infuraSecret).toString('base64')}`
+} : {}
+
 // Helper function to construct a subgraph compiler
 const createCompiler = (
   manifest,
@@ -25,6 +30,7 @@ The IPFS URL must be of the following format: http(s)://host[:port]/[path]`)
         host: url.hostname,
         port: url.port,
         'api-path': url.pathname.replace(/\/$/, '') + '/api/v0/',
+        headers,
       })
     : undefined
 
